### DISCUSSION TITLE
Cdflib available as webassembly

### DISCUSSION THREAD

**Discussion by husayt** [josdejong/mathjs#3052] [Ideas]
https://www.npmjs.com/package/cdflib_wasm

Hi, İ thought this might be an interesting thing to consider

cdflib_wasm is a WebAssembly packaging of the cdflib library as it appears in the presto project.

This library contains routines to compute cumulative distribution functions, inverses, and parameters of the distribution for the following set of statistical distributions:

(1) Beta 
(2) Binomial 
(3) Chi-square 
(4) Noncentral Chi-square 
(5) F 
(6) Noncentral F 
(7) Gamma 
(8) Negative Binomial 
(9) Normal 
(10) Poisson 
(11) Student's t 
(12) Noncentral Student's t

**Comment by josdejong** [comment]
Thanks for sharing @husayt . Good to hear from you, it's quite some time ago 😄 

Maybe it's possible to create a mathjs extension for this library? Or is it simply a matter of importing those function in mathjs?

**Comment by husayt** [comment]
Hi @josdejong, indeed it has been a long long time. I have just dropped an email to you to catch up. Maybe we can take it from there.

**Comment by a-ma72** [comment]
I use 'Math Mode' (https://github.com/CalebJohn/joplin-math-mode) that takes usage of mathjs as plug in for Joplin, but sadly missing statistic functions as listed above. I would appreciate this being part of mathjs.

Regards, Andreas

**Comment by gwhitney** [comment]
Integrating a WASM library into mathjs seems to be more of a general discussion than an actionable issue, so moving to Discussions.

**Comment by kgryte** [comment]
Most of the distributions listed above already exist in [stdlib](https://github.com/stdlib-js/stdlib/tree/ede9ef830c3e6efcdf0f427a1dc651856c41e6ab/lib/node_modules/%40stdlib/stats/base/dists) and, as they are written in JavaScript, can likely be more readily leveraged in Math.js.
